### PR TITLE
Groovy and Kotlin source for micronaut-token-propagation

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

File: guides/micronaut-token-propagation/gateway/groovy/src/test/groovy/example/micronaut/UserControllerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    void testAuthenticatedCanFetchUsername() {
+        when:
+        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials('sherlock', 'password')
+        HttpRequest<?> request = HttpRequest.POST('/login', credentials)
+
+        BearerAccessRefreshToken bearerAccessRefreshToken = client.toBlocking().retrieve(request, BearerAccessRefreshToken)
+
+        String username = client.toBlocking().retrieve(HttpRequest.GET('/user')
+                .header('Authorization', 'Bearer ' + bearerAccessRefreshToken.accessToken), String)
```
Comment (sdelamo) [inline guides/micronaut-token-propagation/gateway/groovy/src/test/groovy/example/micronaut/UserControllerSpec.groovy]: Please, use [HttpRequest.bearerAuth](https://docs.micronaut.io/2.4.2/api/io/micronaut/http/MutableHttpRequest.html#bearerAuth-java.lang.CharSequence-)

File: guides/micronaut-token-propagation/gateway/kotlin/src/test/kotlin/example/micronaut/UserControllerTest.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Test
+    fun testAuthenticatedCanFetchUsername() {
+        val credentials = UsernamePasswordCredentials("sherlock", "password")
+        val request: HttpRequest<*> = HttpRequest.POST("/login", credentials)
+        val bearerAccessRefreshToken = client.toBlocking().retrieve(request, BearerAccessRefreshToken::class.java)
+        val username = client.toBlocking()
+                .retrieve(HttpRequest.GET<Any>("/user")
+                .header("Authorization", "Bearer " + bearerAccessRefreshToken.accessToken), String::class.java)
```
Comment (sdelamo) [inline guides/micronaut-token-propagation/gateway/kotlin/src/test/kotlin/example/micronaut/UserControllerTest.kt]: Please use [`HttpRequest::bearerAuth`](https://docs.micronaut.io/2.4.2/api/io/micronaut/http/MutableHttpRequest.html#bearerAuth-java.lang.CharSequence-)

File: guides/micronaut-token-propagation/intermediate-gateway/groovy/src/test/groovy/example/micronaut/UserControllerSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    void testAuthenticatedCanFetchUsername() {
+        when:
+        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials('sherlock', 'password')
+        HttpRequest<?> request = HttpRequest.POST('/login', credentials)
+
+        BearerAccessRefreshToken bearerAccessRefreshToken = client.toBlocking().retrieve(request, BearerAccessRefreshToken)
+
+        String username = client.toBlocking().retrieve(HttpRequest.GET('/user')
+                .header('Authorization', 'Bearer ' + bearerAccessRefreshToken.getAccessToken()), String)
```
Comment (sdelamo) [inline guides/micronaut-token-propagation/intermediate-gateway/groovy/src/test/groovy/example/micronaut/UserControllerSpec.groovy:39]: Please use [`HttpRequest::bearerAuth`](https://docs.micronaut.io/2.4.2/api/io/micronaut/http/MutableHttpRequest.html#bearerAuth-java.lang.CharSequence-)

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
@burtbeckwith could you check the failing tests?

**Comment by burtbeckwith** [comment]
@sdelamo I manually fixed the 2 GitHub workflows for this guide, and fixed the issue in https://github.com/micronaut-projects/micronaut-guides/pull/826

**Pull request merged**